### PR TITLE
MLP: Add grouping support to neural nets

### PR DESCRIPTION
**Pull request by njayaram2**
JIRA: MADLIB-1149

Changes to support grouping with neural nets. Changes include:
- Standardize independent features by group.
- Use GroupIterationController to iterate.
- Create a temp scaled input table to be used by the
  GroupIterationController
- Create a new UDF that converts a standard deviation value of
  0.0 to 1.0 when standardizing the independent variable.
- Update docs, add more examples.
- Refactor some utility functions code that was used by other
  modules such as svm and elastic_net.
- Update install check test cases scenario for MLP.
- Create a new standardization output table that stores the
  x_mean and x_std per group.
- Add a new method in in_mem_group_control.py_in that returns
  back a specific index's value from the state variable.

Closes #179

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/197/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/200/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/206/

**Comment by njayaram2** [comment]
Jenkins ok to test

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/207/

**Comment by njayaram2** [comment]
Jenkins ok to test

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/208/

**Comment by njayaram2** [comment]
Jenkins ok to test

**Comment by njayaram2** [comment]
Jenkins ok to test

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/210/

**Comment by njayaram2** [comment]
Jenkins ok to test

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/212/

**Comment by njayaram2** [comment]
Jenkins ok to test

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/213/

**Comment by edespino** [comment]
Jenkins ok to test

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/222/

File: src/ports/postgres/modules/convex/mlp_igd.py_in [hunk 1]
```diff
@@ -158,175 +236,257 @@ def mlp(schema_madlib, source_table, output_table, independent_varname,
                 # See design doc for more info
                 span = math.sqrt(6.0 / (fan_in + fan_out))
                 dim = (layer_sizes[i] + 1) * layer_sizes[i + 1]
-                rand = plpy.execute("""SELECT array_agg({span}*(random()-0.5))
+                rand = plpy.execute("""SELECT array_agg({span}*2*(random()-0.5))
```
[the remark this answers is not in the document]
Comment (njayaram2) [reply src/ports/postgres/modules/convex/mlp_igd.py_in]: No, there does not seem to be any particular reason. I think @cooper-sloan had modified it.
It just changes the initial random weights for the coefficients generated, and does not seem to affect the algorithm's convergence all that much.

**Comment by njayaram2** [comment]
Thank you for the comments @orhankislal , will update the PR with relevant changes.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/224/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/228/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/235/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/243/

**Pull request closed without merging**